pkg/util/ovs/fake_ovs.go

```diff
@@ -0,0 +1,447 @@
+package ovs
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ovsFake implements a fake ovs.Interface for testing purposes
+type ovsFake struct {
+	bridge string
+
+	ports map[string]int
+	flows []ovsFlow
+}
+
+// ovsFlow represents an OVS flow
+type ovsFlow struct {
+	table    int
+	priority int
+	created  time.Time
+	cookie   string
+	fields   []ovsField
+	actions  string
+}
+
+type ovsField struct {
+	name  string
+	value string
+}
+
+const (
+	minPriority     = 0
+	defaultPriority = 32768
+	maxPriority     = 65535
+)
+
+// NewFake returns a new ovs.Interface
+func NewFake(bridge string) Interface {
+	return &ovsFake{bridge: bridge}
+}
+
+func (fake *ovsFake) AddBridge(properties ...string) error {
+	fake.ports = make(map[string]int)
+	fake.flows = make([]ovsFlow, 0)
+	return nil
+}
+
+func (fake *ovsFake) DeleteBridge() error {
+	fake.ports = nil
+	fake.flows = nil
+	return nil
+}
+
+func (fake *ovsFake) ensureExists() error {
+	if fake.ports == nil {
+		return fmt.Errorf("no bridge named %s", fake.bridge)
+	}
+	return nil
+}
+
+func (fake *ovsFake) GetOFPort(port string) (int, error) {
+	if err := fake.ensureExists(); err != nil {
+		return -1, err
+	}
+
+	if ofport, exists := fake.ports[port]; exists {
+		return ofport, nil
+	} else {
+		return -1, fmt.Errorf("no row \"%s\" in table Interface", port)
```

Comment:
Yeah... in this case I was trying to get exactly the same error message as the command-line returns, and didn't want to rely on %q using double-quotes rather than single-quotes. But there's no real reason we need to guarantee exactly the same error message.